test/test-unix-socket.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+var tempfile = require('tempfile');
+var format = require('util').format;
+var test = require('tap').test;
+var got = require('../');
+var server = require('./server.js');
+var s = server.createServer();
+
+var socketPath = tempfile('.socket');
+
+s.on('/', function (req, res) {
+	res.end('ok');
+});
+
+test('setup', function (t) {
+	s.listen(socketPath, function () {
+		t.end();
+	});
+});
+
+test('request via unix socket', function (t) {
+	// borrow unix domain socket url format from request module
+	var url = format('http://unix:%s:%s', socketPath, '/');
```

Comment:
Can you add a test to make sure `unix:%s:%s', socketPath, '/'` works too? We support protocol-less URLs for normal URLs.